### ISSUE TITLE
net: implement closeRead and closeWrite on Plan 9

### ISSUE THREAD

**Issue by 0intro** [golang/go#17906]
We should implement closeRead and closeWrite on Plan 9, so the following tests could be enabled:

- TestCloseRead
- TestCloseWrite
- TestClientWriteShutdown (net/http package, see #7237)

**Comment by bradfitz** [comment]
I'm marking this as Unplanned but David, feel free to implement whenever you want, regardless of release cycle, as long as it only touches plan9 files or removes t.Skip calls.

**Comment by bradfitz** [comment]
@0intro, does Plan 9 even support half closed TCP connections?

**Comment by fhs** [comment]
@bradfitz Nope. There was some [9fans discussion](https://9fans.topicbox.com/groups/9fans/T65d8e487583176c6-Mf42b3d6653587dd41be5f4ef/9fans-adding-tcp-half-duplex-close) some years ago about adding support for it in the Plan 9 kernel, and there is a [kernel patch](https://9p.io/sources/patch/tcp-halfduplex-close/) but it hasn't been merged yet.

**Comment by 0intro** [comment]
This is interesting. I didn't remember this change. I'll try it and eventually merge it.
